x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -669,6 +669,8 @@ export class TaskRunner<
       schedule: taskSchedule,
     } = this.taskInstance;
 
+    this.logger = createTaskRunnerLogger({ logger: this.logger, tags: [ruleId, this.ruleType.id] });
```

Comment:
Checking some existing cluster, I can see these values already being used with the `rule-run-failed` tag.  So, perfect!  Which I'm guessing might be out of the task_runner logic, didn't look.  But made me think, we can probably use this more widely, like in our HTTP endpoints that target specific rules.